### PR TITLE
Fix inconsistency between poisson_nll_loss rfc and implementation

### PR DESCRIPTION
**Pull request by LyndonKong**
Follow the suggesstions of our kind reviewer, we update our rfc doc. Refer to this [pr](https://github.com/PaddlePaddle/Paddle/pull/51117)

### CODE REVIEW

File: rfcs/APIs/20230301_api_design_for_poissonnllloss.md [hunk 1]
```diff
@@ -46,20 +46,20 @@ PyTorch：PyTorch 支持 torch.nn.poissonNLLLoss 和 torch.nn.functional.Poisson
 ```python
 def poisson_nll_loss(
     input: Tensor,
-    target: Tensor,
+    label: Tensor,
```
Comment (GGBond8488) [inline rfcs/APIs/20230301_api_design_for_poissonnllloss.md:49]: 这里是torch的定义吧，不用改成label

File: rfcs/APIs/20230301_api_design_for_poissonnllloss.md [hunk 2]
```diff
@@ -46,20 +46,20 @@ PyTorch：PyTorch 支持 torch.nn.poissonNLLLoss 和 torch.nn.functional.Poisson
...
     log_input: bool = True,
     full: bool = False,
     size_average: Optional[bool] = None,
     epsilon: float = 1e-8,
     reduce: Optional[bool] = None,
     reduction: str = "mean",
 ) -> Tensor:
-    if has_torch_function_variadic(input, target):
+    if has_torch_function_variadic(input, label):
```
Comment (GGBond8488) [inline rfcs/APIs/20230301_api_design_for_poissonnllloss.md:57]: 同上，其他torch的也一样

File: rfcs/APIs/20230301_api_design_for_poissonnllloss.md (hunk 1 above)
Comment (LyndonKong) [reply rfcs/APIs/20230301_api_design_for_poissonnllloss.md]: 在新提交里已经修改

**Review by GGBond8488 — APPROVED** [review]
LGTM

**Review by luotao1 — APPROVED** [review]
